indexer/boundary_boxes.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "geometry/bounding_box.hpp"
+#include "geometry/calipers_box.hpp"
+#include "geometry/diamond_box.hpp"
+#include "geometry/point2d.hpp"
+
+#include "base/visitor.hpp"
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace indexer
+{
+struct BoundaryBoxes
```

Comment:
final? (здесь и в других аналогичных местах)

Reply:
Well, I generally don't think why? From my point of view we need to mark classes as "final" only when we don't want them to be inherited. Not the case here.

Reply:
Ok. It's up to you. 

I prefer using the key word "final" and the key word "const" in the same manner. Form my point of view if we don't change a value we mark it as a "const", if we don't inherit from a class I mark it as "final". 

You wrote: "we need to mark classes as "final" only when we don't want them to be inherited". But  are you sure that you what a class without a virtual destructor to be inherited?  (My comment comes to all new classes in your PR.)

Reply:
I think that absence of a virtual destructor is not a big deal and can be easily fixed, right?

When I say "we don't want them to be inherited" I mean that there're some serious reasons against usage of this class as a base class, and absence of virtual destructor is not the case. So, to me, "final" means "Beware! Author of the class designed it to be a final class in the hierarchy, and there should be some complex invariants that may be violated by subclasses".

Reply:
So I don't want to use "final" keyword only because we don't have subclasses right now.

Reply:
Of course, we can discuss the semantic of the "final" keyword (possibly, with other members of our team), and update the style guide, to prevent disputes in the future.

Reply:
What does Meyers think about this?

Reply:
For example, here I found an example where final is *really* useful: https://akrzemi1.wordpress.com/2012/09/30/why-make-your-classes-final/

Reply:
Ok. I see you point of view. Probably it's worth making a meeting one day to agree about using the word final. It would nice if we use "final" key word in the same manner. As for Meyers he wrote about it nothing significant in his "Effective Modern C++ 2014.pdf".

Reply:
We can start to follow the following rule: "class should have either virtual dtor or marked as final". WDYT?

Reply:
Also, take a look at http://foonathan.net/blog/2016/05/27/final.html - if the class don't have virtual functions, there's no need to mark it as final too.

Reply:
Thanks for the links. Especially for the second one.